<a href="https://colab.research.google.com/github/AchmadFauziTrinanda/digitalskola_ds11/blob/master/Intermediate_Dataframe_I.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#***Intermediate Dataframe I***

by : Achmad Fauzi Trinanda

In [ ]:
import pandas as pd
import numpy as np

# ***Load Dataset 1***

In [ ]:
df1 = pd.read_csv("https://raw.githubusercontent.com/densaiko/data_science_learning/main/dataset/data.csv")
df1.head()

Unnamed: 0  employee_id  ... avg_training_score is_promoted
0           0        65438  ...               49.0           0
1           1        65141  ...               60.0           0
2           2         7513  ...               50.0           0
3           3         2542  ...               50.0           0
4           4        48945  ...               73.0           0

[5 rows x 14 columns]

# ***Load Dataset 2***

In [ ]:
df2 = pd.read_csv("https://raw.githubusercontent.com/densaiko/data_science_learning/main/dataset/data_2.csv")
df2.head()

Unnamed: 0  employee_id  ... avg_training_score is_promoted
0       33498        32835  ...               56.0           0
1       31110        72340  ...               62.0           0
2       47412        52336  ...               54.0           0
3        9900        75679  ...               59.0           0
4       39275        72476  ...               57.0           0

[5 rows x 14 columns]

# ***Full Outer Join***

In [ ]:
df3 = df1.merge(df2,on='employee_id',how='outer')
df3.head()

Unnamed: 0_x  employee_id  ... avg_training_score_y is_promoted_y
0             0        65438  ...                  NaN           NaN
1             1        65141  ...                 60.0           0.0
2             2         7513  ...                  NaN           NaN
3             3         2542  ...                  NaN           NaN
4             3         2542  ...                  NaN           NaN

[5 rows x 27 columns]

Check Dimension


In [ ]:
df3.shape

(59858, 27)

In [ ]:
df_clean = df3.loc[(df3["awards_won_y"]).isnull()]
df_clean.head()

Unnamed: 0_x  employee_id  ... avg_training_score_y is_promoted_y
0             0        65438  ...                  NaN           NaN
2             2         7513  ...                  NaN           NaN
3             3         2542  ...                  NaN           NaN
4             3         2542  ...                  NaN           NaN
5             4        48945  ...                  NaN           NaN

[5 rows x 27 columns]

In [ ]:
df_clean.shape

(54022, 27)

Terdapat perbedaan setelah mencari data yang "null" jumlah row berkurang dari 59858 menjadi 54022 row

# ***Mengambil Dataset 1 Saja***

In [ ]:
df_new = df_clean.iloc[:,0:14]
df_new.head()

Unnamed: 0_x  employee_id  ... avg_training_score_x is_promoted_x
0             0        65438  ...                 49.0             0
2             2         7513  ...                 50.0             0
3             3         2542  ...                 50.0             0
4             3         2542  ...                 50.0             0
5             4        48945  ...                 73.0             0

[5 rows x 14 columns]

#***Mengubah Nama Kolom***

In [ ]:
df_new.rename(columns = {'Unnamed: 0_x': 'Unnamed: 0',
                         'department_x':'department',
                         'region_x':'region',
                         'education_x' :'education',
                         'gender_x': 'gender',
                         'recruitment_channel_x' : 'recruitment_channel',
                         'no_of_trainings_x' : 'no_of_trainings',
                         'age_x' : 'age',
                         'previous_year_rating_x' : 'previous_year_rating',
                         'length_of_service_x' : 'length_of_service',	
                         'awards_won_x' : 'awards_won',	
                         'avg_training_score_x' : 'avg_training_score',
                         'is_promoted_x' : 'is_promoted'}).head()

Unnamed: 0  employee_id  ... avg_training_score is_promoted
0           0        65438  ...               49.0           0
2           2         7513  ...               50.0           0
3           3         2542  ...               50.0           0
4           3         2542  ...               50.0           0
5           4        48945  ...               73.0           0

[5 rows x 14 columns]

In [ ]:
df_new.shape

(54022, 14)